primitives/primproc/batchprimitiveprocessor.cpp

```diff
@@ -1991,15 +1988,16 @@ void BatchPrimitiveProcessor::writeProjectionPreamble()
         {
             *serialized << (uint8_t) 1;
             *serialized << lbidForCP;
-            if (hasBinaryColumn)
+            if (UNLIKELY(hasWideColumnOut))
             {
-                *serialized << (uint8_t) 16; // width of min/max value
-                *serialized << (unsigned __int128) bigMinVal;
-                *serialized << (unsigned __int128) bigMaxVal;
+                // PSA width
+                *serialized << (uint8_t) wideColumnWidthOut;
+                *serialized << min128Val;
+                *serialized << max128Val;
```

Comment:
These should be casted to unsigned before serializing, as the receiving end is deserializing uint128_t.